### PR TITLE
[121] Add projects to a freak

### PR DESCRIPTION
**Pull request by salcudeann**
## Related

[[42] Add Projects to Freak](https://trello.com/c/RQBdanH4/121-add-projects-to-freak)

## Steps to Test or Reproduce

*To be resolved*

### CODE REVIEW

File: app/models/freak.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 class Freak < ApplicationRecord
   has_one :photo, as: :imageable, dependent: nil
+  has_many :projects, as: :proj, dependent: nil
```
Comment (alex-damian-negru) [inline app/models/freak.rb:5]: Please avoid incomplete terms (e.g. `proj`) as they can be unclear. I don't think the `as` or `dependent` options are needed here either.

File: app/models/project.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class Project < ApplicationRecord
+  belongs_to :proj, polymorphic: true
```
Comment (alex-damian-negru) [inline app/models/project.rb:4]: Why is this polymorphic & referencing itself? o_O

File: db/migrate/20210802074146_create_freaks.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ def change
     create_table :freaks do |t|
       t.string :name
       t.string :description
-      t.string :photo
+      t.string :email
```
Comment (alex-damian-negru) [inline db/migrate/20210802074146_create_freaks.rb:8]: Avoid changing old migrations. All new DB changes should happen in new migrations.

That aside, why rename this to `email`?

File: db/migrate/20210811084757_create_projects.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class CreateProjects < ActiveRecord::Migration[6.1]
+  def change
+    create_table :projects do |t|
+      t.string :name
+      t.string :description
+      t.references :proj, polymorphic: true
```
Comment (alex-damian-negru) [inline db/migrate/20210811084757_create_projects.rb:8]: What's the purpose behind this?

**Review by alex-damian-negru — CHANGES_REQUESTED** [review]

**Comment by alex-damian-negru** [comment]
@salcudeann please rebase this.

**Comment by salcudeann** [comment]
> @salcudeann please rebase this.

I think i should close thir PR and create a new one. I started with some mistakes and i think it would be better if i make a new branch for creating a project.

**Pull request closed without merging**